src/ESA.cpp

```diff
@@ -0,0 +1,232 @@
+#include <omp.h>
+#include "ESA.hpp"
+#include <sstream>
+#include <iomanip>
+#include <cmath>
+
+using namespace std;
+
+// -----------------------------------------------------------------
+// ESA class
+// -----------------------------------------------------------------
+
+ESA::ESA(const StlsInput& input) : Stls(input) {
```

Comment:
Since this constructor is not doing anything except calling the Stls constructor, remove it from here and move it to the header file esa.hpp (see my comment in esa.hpp)